### ISSUE TITLE
BUG: values that are not dates can get formatted as if they were

### ISSUE THREAD

**Issue by oscarssanchez** [10up/ElasticPress#2769]
**Describe the bug**

Looks like https://github.com/10up/ElasticPress/blob/18a0d3b0797d258c99e519cae87a84a4fa3b6faf/includes/classes/Indexable.php#L711 is treating some values that are not dates as if they were, later causing issues with ElasticSearch. If we try to prepare the following value: 5479516sunt it gets formatted as a valid date value 107039-03-10

<img width="1530" alt="Screen Shot 2022-05-17 at 9 00 13" src="https://user-images.githubusercontent.com/31049169/168829228-b43a5dab-fb93-4c05-b036-31fa9130dd99.png">

**Steps to Reproduce**

1. Have a post with post meta key test and value 5479516sunt
2. Run an indexing operation
3. See instead of default date values 1970-01-01 is 107039-03-10

**Expected behavior**

**Screenshots**

**Environment information**
 - Device: 
 - OS: 
 - Browser and version: 
 - WordPress version: 
 
 - ElasticPress version: 
 - Elasticsearch version: 
 - Where do you host your Elasticsearch server? 
 - What ElasticPress features do you have currently active? 
 - Plugins and version: 
 - Theme and version: 
 
 - <details><summary>Site Health Info:</summary>
   
   </details>

**Additional context**

**Comment by felipeelia** [comment]
We will need to cap out the max Year there, preferably the same way Elasticsearch does. Docs related: https://www.elastic.co/guide/en/elasticsearch/reference/7.17/mapping-date-format.html#strict-date-time

**Comment by felipeelia** [comment]
https://www.elastic.co/guide/en/elasticsearch/reference/8.2/mapping-date-format.html
https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
> Year: The count of letters determines the minimum field width below which padding is used. If the count of letters is two, then a reduced two digit form is used. For printing, this outputs the rightmost two digits. For parsing, this will parse using the base value of 2000, resulting in a year within the range 2000 to 2099 inclusive. If the count of letters is less than four (but not two), then the sign is only output for negative years as per SignStyle.NORMAL. Otherwise, the sign is output if the pad width is exceeded, as per SignStyle.EXCEEDS_PAD.

TL;DR 2099

(props to @tott)
